app/uk/gov/hmrc/bindingtariffclassification/repository/SearchMapper.scala

```diff
@@ -114,6 +114,14 @@ class SearchMapper @Inject()(appConfig: AppConfig) {
     "$in" -> JsArray(values.toSeq.map(writes.writes)))
   )
 
+  private def inArrayOrNone[T](values: TraversableOnce[T])(implicit writes: Writes[T]):JsValue = {
+    if (values.exists(_ == "none")) JsNull else
+    if (values.exists(_ == "some")) Json.obj("$ne" -> JsNull) else
+      JsObject(
+        Map("$in" -> JsArray(values.toSeq.map(writes.writes)))
+      )
+  }
```

Comment:
Interesting. So the queueId search currently supports the strings "some" or "none" to search for cases assigned to any queue or no queue (gateway). Now that we are supporting a sequence of queueId's, we need to consider sequences such as `Set("none", "5")` or `Set("some", "5")` or even `Set("some", "none", ...)`.

For `Set("some", "5")` we can't easily combine a `{ $ne: null }` with an `$in` query, but this is unnecessary since "some" will include "5" anyway, so in this case the code you have is OK.

For `Set("none", "5")` the code currently would just treat this as `Set("none")` which I don't think is correct. MongoDB seems happy with a query such as `{ queueId: { $in: [ null, "5" ] } }` so we should be able to improve this.

For `Set("none", "some")` ideally we wouldn't have any filter on the queueId so we need to handle this condition prior to calling this method.

```suggestion
  private def inArrayOrNone[T](values: TraversableOnce[T])(implicit writes: Writes[T]): JsObject = {
    values match {
      case _ if values.exists(_ == "some") =>
        Json.obj("$ne" -> JsNull)
      case _ if values.exists(_ == "none") =>
        JsObject(Map(
          "$in" -> JsArray(JsNull :: values.toList.filterNot(_ == "none").map(writes.writes))
        ))
      case _ =>
        inArray(values)
    }
  }
```

Reply:
Thanks that makes sense! We have updated the code and written couple of new test cases to cover this logic.